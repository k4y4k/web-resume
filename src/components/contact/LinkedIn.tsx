import * as React from 'react'

interface LinkedInTypes {
  username?: string
}

const LinkedIn = ({
  username = 'Error: No LinkedIn',
}: LinkedInTypes): JSX.Element => {
  if (username === 'Error: No LinkedIn') return <p>{username}</p>

  return (
    <p>
      <a href={`https://linkedin.com/in/${username}`} target='_blank'>
        {username}
      </a>
    </p>
  )
}

export default LinkedIn
