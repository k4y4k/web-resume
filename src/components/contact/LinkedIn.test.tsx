import * as React from 'react'
import { screen, render } from '@testing-library/react'
import LinkedIn from './LinkedIn'

describe('<LinkedIn />', () => {
  test('handles no data', () => {
    render(<LinkedIn username='' />)

    const linkedin = screen.getByText('Error: No LinkedIn')
    expect(linkedin).not.toBeFalsy()
  })

  test('links to profile', () => {
    render(<LinkedIn username='exampledin' />)

    const linkedin = screen.getByText('exampledin')
    expect(linkedin).not.toBeFalsy()
    expect(linkedin).toHaveAttribute(
      'href',
      'https://linkedin.com/in/exampledin'
    )
  })

  test('opens in new tab', () => {
    render(<LinkedIn username='exampledin' />)

    const linkedin = screen.getByText('exampledin')
    expect(linkedin).not.toBeFalsy()
    expect(linkedin).toHaveAttribute('target', '_blank')
  })

  test.todo('has cute little logo')
})
