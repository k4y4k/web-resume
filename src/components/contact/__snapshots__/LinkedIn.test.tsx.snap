// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkedIn /> handles no data 1`] = `null`;

exports[`<LinkedIn /> handles null data 1`] = `null`;

exports[`<LinkedIn /> links to profile in new tab 1`] = `
<a
  data-testid="contactLinkedin"
  href="https://example.com"
  rel="noreferrer"
  target="_blank"
>
  <svg
    fill="none"
    height="1em"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"
    />
    <rect
      height="12"
      width="4"
      x="2"
      y="9"
    />
    <circle
      cx="4"
      cy="4"
      r="2"
    />
  </svg>
   
  exampledin
</a>
`;
